### ISSUE TITLE
[Bug]: Number literals are interpreted differently depending on locale

### ISSUE THREAD

**Issue by ladipro** [dotnet/msbuild#8798]
### Issue Description

Passing literals with dots and commas to property functions results in differently evaluated projects depending on locale / regional settings under which the MSBuild process runs.

### Steps to Reproduce

```xml
<Project>
  <Target Name="Build">
    <Message Text="$([MSBuild]::Add('1,2', 1))"/>
  </Target>
</Project>
```

### Expected Behavior

The `Message` task should print `13` regardless of locale.

### Actual Behavior

English/invariant: `13`

Switch **Regional format** to one that uses comma for decimal separator, for example French. The task prints `2,2`.

### Analysis

This is caused by calling the default overloads of `double.Parse` in evaluation.

Doesn't look like a recent regression but the .NET Framework-bundled MSBuild doesn't seem to have this bug.

### Versions & Configurations

MSBuild version 17.7.0-preview-23267-02+fe79798bd for .NET Framework

**Comment by jrdodds** [comment]
The example in the steps to reproduce is resolved by PR #8710 for issue #8698 but that may not be a complete resolution of the locale issue.

Please assign this issue to me. I'll check for other usages of the default overloads of `double. Parse`.

**Comment by jrdodds** [comment]
PR #8710 has been merged and contains all the changes from PR #8853. This issue has been completed by PR #8710.

**Comment by ladipro** [comment]
Thank you!
